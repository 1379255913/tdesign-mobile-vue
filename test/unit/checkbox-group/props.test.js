/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { mount } from '@vue/test-utils';
import CheckboxGroup from '../../../src/checkbox-group';
import { h } from 'vue';

describe('CheckboxGroup', () => {
  describe(':props', () => {
    
    it('disabled', () => {
      const fn = jest.fn();
      const wrapper = mount(CheckboxGroup, {
        propsData: {
          disabled: true,
          onClick: fn,
        },
      });
      wrapper.trigger('click');
      expect(fn).not.toHaveBeenCalled();
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 是否禁用组件: false
    it('disabled', () => {
      const wrapper = mount(CheckboxGroup, {
        propsData: {"disabled":true},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 支持最多选中的数量: undefined
    it('max', () => {
      const wrapper = mount(CheckboxGroup, {
        propsData: {"max":"undefined"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 统一设置内部复选框 HTML 属性: 
    it('name', () => {
      const wrapper = mount(CheckboxGroup, {
        propsData: {"name":""},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 以配置形式设置子元素。示例1：`['北京', '上海']` ，示例2: `[{ label: '全选', checkAll: true }, { label: '上海', value: 'shanghai' }]`。checkAll 值为 true 表示当前选项为「全选选项」: []
    it('options', () => {
      const wrapper = mount(CheckboxGroup, {
        propsData: {"options":"[]"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });// 选中值: []
    it('value', () => {
      const wrapper = mount(CheckboxGroup, {
        propsData: {"value":"[]"},
      });
      expect(wrapper.wrapperElement).toMatchSnapshot();
    });
  });
});