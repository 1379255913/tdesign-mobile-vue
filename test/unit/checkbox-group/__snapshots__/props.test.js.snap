// Vitest Snapshot v1

exports[`CheckboxGroup > :props > max 1`] = `
<div
  class="t-checkbox-group"
>
  
  
</div>
`;

exports[`CheckboxGroup > :props > name 1`] = `
<div
  class="t-checkbox-group"
>
  
  
</div>
`;

exports[`CheckboxGroup > :props > value 1`] = `
<div
  class="t-checkbox-group"
>
  
  
</div>
`;
